boards/nxp/frdm_mcxn236/frdm_mcxn236.dtsi

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "frdm_mcxn236-pinctrl.dtsi"
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	aliases{
+		led0 = &red_led;
+		led1 = &green_led;
+		led2 = &blue_led;
+		sw0 = &user_button_2;
+		sw1 = &user_button_3;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		green_led: led_1 {
+			gpios = <&gpio4 19 GPIO_ACTIVE_LOW>;
+			label = "Green LED";
+		};
+		blue_led: led_2 {
+			gpios = <&gpio4 17 GPIO_ACTIVE_LOW>;
+			label = "Blue LED";
+		};
+		red_led: led_3 {
+			gpios = <&gpio4 18 GPIO_ACTIVE_LOW>;
+			label = "Red LED";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button_2: button_0 {
+			label = "User SW2";
+			gpios = <&gpio0 20 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+			zephyr,code = <INPUT_KEY_WAKEUP>;
+		};
+		user_button_3: button_1 {
+			label = "User SW3";
+			gpios = <&gpio0 6 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+	};
+};
+
+&flexcomm3_lpspi3 {
+	pinctrl-0 = <&pinmux_flexcomm3_lpspi>;
+	pinctrl-names = "default";
+};
+
+&flexcomm5_lpi2c5 {
+	pinctrl-0 = <&pinmux_flexcomm5_lpi2c>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+};
+
+&flexcomm2_lpuart2 {
+	current-speed = <115200>;
+	pinctrl-0 = <&pinmux_flexcomm2_lpuart>;
+	pinctrl-names = "default";
+};
+
+&flexcomm4_lpuart4 {
+	current-speed = <115200>;
+	pinctrl-0 = <&pinmux_flexcomm4_lpuart>;
+	pinctrl-names = "default";
+};
+
+/*
+ * MCXN236 board uses OS timer as the kernel timer
+ * In case we need to switch to SYSTICK timer, then
+ * replace &os_timer with &systick
+ */
+&os_timer {
+	status = "okay";
+};
+
+&flash {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(64)>;
+		};
+		/* Note slot 0 has one additional sector,
+		 * this is intended for use with the swap move algorithm
+		 */
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_K(480)>;
+		};
+		slot1_partition: partition@80000 {
+			label = "image-1";
+			reg = <0x0088000 DT_SIZE_K(472)>;
+		};
+		/* storage_partition is placed in WINBOND flash memory*/
```

Comment:
It doesn't seem like we have the winbond flash memory defined for this board yet in the devicetree, so we should probably drop this comment for the initial board enablement